lib/get-specs-from-issue.js

```diff
@@ -2,38 +2,81 @@ module.exports = getSpecsFromIssue;
 
 function getSpecsFromIssue(context, issueNumber) {
   console.log('started getting specs out of issue');
-  let issue = context.github.issues;
-  let specs = [];
+  const repo = context.repo();
 
   return Promise.resolve()
-  .then(() => Promise.all([
-    context.github.issues.get(...repo, issueNumber),
-    context.github.issues.listComments(...repo, issueNumber)
-  ]))
-  .then(data => {
-    const { body, comments } = data;
-    const allCommentsBody = comments
-      .map((comment) => comment.body)
-      .push(body);
-
-    pushSpecs(specs, allCommentsBody);
-
-    return specs;
-  });
+    .then(() => Promise.all([
+      context.github.issues.get(...repo, issueNumber),
+      context.github.issues.listComments(...repo, issueNumber)
+    ]))
+    .then(data => {
+      const { issueBody, issueComments } = data;
+
+      return issueComments
+        .map(comment => comment.body)
+        .push(issueBody)
+        .join('\n');
+    })
+    .then(issueCompleteText => getSpecsArray(issueCompleteText))
+    .then(specsArray => specNamesFromMarkdownList(specsArray));
 }
 
-function pushSpecs(specs, body) {
-  body.toLowerCase();
-  let posiotionOfSpecsList = body.indexOf('## specs');
-  body = body.slice(posiotionOfSpecsList);
-  let endPosiotionOfSpecsList = body.indexOf('##');
-  body = body.slice(0, endPosiotionOfSpecsList);
-
-  let bullets = body.split('-');
-  bullets.forEach(bullet => {
-    bullet.trim();
-    if (bullet.startsWith('it')) {
-      specs.push(bullet);
+function getSpecsArray(completeText) {
+  const specsBlockStartRegex = /^\s*?(#+)\s*?specs/gi
+  const specsArray = [];
+  var isInsideSpecsBlock = false;
+  var specsBlockTopLevelHeader;
+
+  completeText.split('\n').forEach(line => {
+    if (!isInsideSpecsBlock) {
+      const lineMatchResult = specsBlockStartRegex.exec(line);
+      if (lineMatchResult) {
+        isInsideSpecsBlock = true;
+        specsBlockTopLevelHeader = lineMatchResult[1];
+      }
+    } else {
+      const isHigherLevelMarkdownTitle = isMarkdownTitle(line) &&
+        !isMarkdownTitleDepthValid(line, specsBlockTopLevelHeader);
+
+      const isLineValid = isEmptyLine(line) ||
+        isMarkdownListItem(line) ||
+        isMarkdownTitle(line) &&
+          isMarkdownTitleDepthValid(line, specsBlockTopLevelHeader);
+
+      if (isHigherLevelMarkdownTitle) {
+        isInsideSpecsBlock = false;
+      } else if (isLineValid) {
+        specsArray.push(line);
+      }
     }
   });
+
+  return specsArray;
+}
+
+function isMarkdownTitle(line) {
+  return line.startsWith('#');
+}
+
+function isMarkdownTitleDepthValid(title, rootTitlePrefix) {
+  return !title.startsWith(rootTitlePrefix)
+} 
+
+function isMarkdownListItem(line) {
+  const markdownListItemRegex = /^\s*?[-\*](?:.*)/g
+  return markdownListItemRegex.test(line);
+}
+
+function isEmptyLine(line) {
+  return line.trim() === '';
```

Comment:
The trim here is in the wrong place. you should trim the line at the start of your run on the lines and call all the help functions (`isMarkdownTitle`, `isMarkdownTitleDepthValid`, `isMarkdownListItem`, and `isEmptyLine`) with the trimmed line already. In markdown, titles & lists can be indented and it shouldn't effect the parsing with the current implementation